model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/RealmCacheSession.java

```diff
@@ -495,13 +495,13 @@ public void evictRealmOnRemoval(RealmModel realm) {
 
     @Override
     public ClientModel addClient(RealmModel realm, String clientId) {
```

Comment:
What about this method and the other client-specific methods? Should we introduce a `ClientCacheSession` class and move this functionality there?

Reply:
Yup, in the long term, this should be the way. This will be handled as part of the caching layer in store.x in some of the follow-up PR